src/executable.js

```diff
@@ -41,4 +41,16 @@ export class Executable extends mix(NamedDefaultableInMemoryEntity).with(Runtime
     getFlavorByConfig(config) {
         return config ? this.getFlavorByName(config.name) : this.defaultFlavor;
     }
+
+    getFlavorByApplicationVersion(version) {
+        const filteredFlavors = this.flavors.filter((flavor) => {
+            const { supportedApplicationVersions } = flavor.toJSON();
```

Comment:
Let's avoid calling `toJSON()` if not absolutely necessary. We can use
```javascript
const supportedApplicationVersions = flavor.prop("supportedApplicationVersions");
```
instead.